lib/erb_lint/formatters/json_formatter.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module ERBLint
+  module Formatters
+    class JSONFormatter
+      RUBOCOP_MESSAGE_PATTERN = /\A(?<linter>\S+): (?<message>.+)\z/.freeze
+
+      def initialize
+        @result = {
+          metadata: metadata_hash,
+          files: [],
+          summary: {offense_count: 0, inspected_file_count: 0}
+        }
+      end
+
+      def file_completed(relative_filename, runner)
+        @result[:summary][:offense_count] += runner.offenses.count
+        @result[:summary][:inspected_file_count] += 1
+        @result[:files] = hash_for_file(relative_filename, runner)
```

Comment:
We are reporting the results only for the last processed file, and I think it could be better add the files to the final report only if offenses are present.

```suggestion
        if runner.offenses.count.positive?
          @result[:summary][:offending_file_count] += 1
          @result[:files] << hash_for_file(relative_filename, runner)
        end
```

Reply:
Oh, careless mistake ...

And I agree, we should only report files with issues. Iirc, the check is currently not necessary because files without offenses do not reach this code anyways. But to record the number of inspected files this would change.